### ISSUE TITLE
Initialization race condition can lead to empty editor (with a workaround)

### ISSUE THREAD

**Issue by codingyourlife** [tinymce/tinymce-react#232]
**What is the current behavior?**
We have a profile page with a short description and a long description each using the @tinymce/tinymce-react richtext editor. The editor is filled with the delayed info from the database after an axios backend call. This works fine until I reload the page.

After a Ctrl+R reload everything gets really weird:
- The editor is empty but I can still save the old value. When I display the value above the editor it IS there.
- When I type a new text the old value gets overwritten on save so the state updates correctly
- When I click in the address bar and press enter the behavior is the same and the editor is still empty if it came once into this error state
- When I change to a different page and enter the profile settings page again it is also still in this error state
- When I just press save with the empty editor in error state it "repairs" and displays the old value until I press reload again
- The issue is the same if I use value or initialValue to prefill my editor
- Again because this is important: I can display the expected value above the editor but the editor is empty. The variable holds the real value but the editor is not accepting it (because it is not done initializing or interrupted or something)

**Please provide the steps to reproduce**
This is clearly a race condition and I don't think it will be easy to reproduce on codesandbox. But we had it well reproduceable.

**What is the expected behavior?**
Should show the initial value even if it updates few times before editor is fully initialized.

**Which versions of TinyMCE, and which browser / OS are affected by this issue? Did this work in previous versions of TinyMCE or `tinymce-react`?**
I'm on Windows 10 with latest Firefox 86.0.
node version v14.15.3.
@tinymce/tinymce-react version ^3.10.3
I'm not sure if this issue is new but there seems to be at least this related issue. I didn't see any error in the console tough. [https://github.com/tinymce/tinymce-react/issues/208](https://github.com/tinymce/tinymce-react/issues/208)

**Initialization**
Here is how I initialize my Editor in case this is important (note that we use wordcount paste plugins):
```
height: 300,
menubar: false,
toolbar:
	"undo redo | bold italic | bullist numlist | removeformat",
plugins: "wordcount paste",
forced_root_block: "",
paste_as_text: true,
branding: false,
entity_encoding: "raw",
```

**Now to my workaround**
Basically I use the onInit event to wait until the editor is ready to receive the value. Until then I pass in undefined. Here a code sample in a reduced form to my code that should make the usage clear. Sorry in case there are typos etc. it's to illustrate the concept.
```
const TextEditor = ({
	value,
	setValue
}) => {
    const [editorIsInitialized, setEditorIsInitialized] = useState(false);
    
    return <Editor
	    onInit={() => {
		    if (!editorIsInitialized) {
			    setEditorIsInitialized(true); //this fixes the problem that on reload the editor can be empty
		    }
	    }}
	    initialValue={editorIsInitialized ? value : undefined} //if initialValue is not working try value
	    onEditorChange={setValue}
    />
}
```

**Comment by tiny-james** [comment]
I've made an update 3.11.0 which allows for the `initialValue` to be set asynchronously.

See here for an example:
https://codesandbox.io/s/tinymcereact-async-initialvalue-1i0us
 
Let me know if you have any problems. If I don't hear anything I'll close this issue in the next day or so.

**Comment by tiny-james** [comment]
Closing as "fixed"
